### PR TITLE
ISNM: Shadows of the Mind

### PR DESCRIPTION
**Pull request by CatsEyeXI**
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I understand I should leave resolving conversations to the LandSandBoat team so that reviewers won't miss what was said.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?
Add ISNM: Shadows of the Mind

The primary gimmick in this fight, is that the Phantom Puk will periodically use `Boreas Mantle` to spawn clones, with much less HP, but with the same HP% as Phantom Puk.

**Immunities**: 

- requiem    1024
- sleep          4096
- gravity             2
- silence           16
- paralyze        20

**Sources**:
Captures and Video from Captures Discord

- https://youtu.be/Pbdy8edAkRU
- https://drive.google.com/open?id=1M4D4WLg4T_enfbyNPPu_RTu7VDo5T5Pl
- https://discord.com/channels/443544205206355968/609478860593037395/617535963035336704
- https://www.bg-wiki.com/ffxi/Shadows_of_the_Mind

## Steps to test these changes

- !zone Jade Sepulcher
- !addkeyitem SECRET_IMPERIAL_ORDER
- Enter ISNM
- Engage Boss, wait a minute, clones will spawn, 30 seconds later, they will despawn
- Defeat Boss, open crate, get loot.

### CODE REVIEW

**Comment by WinterSolstice8** [comment]
google drive link is for "call to arms"

File: sql/mob_groups.sql [hunk 1]
```diff
@@ -3271,6 +3271,7 @@ INSERT INTO `mob_groups` VALUES (14,6957,67,'Mamool_Ja_Profligate',0,128,0,0,0,9
 INSERT INTO `mob_groups` VALUES (15,6958,67,'Mamool_Ja_Unseen',0,128,0,0,0,99,99,0);
 INSERT INTO `mob_groups` VALUES (16,6959,67,'Mamool_Ja_Flamerearer',0,128,0,0,0,99,99,0);
 INSERT INTO `mob_groups` VALUES (17,6960,67,'Scorched-snout_Wyvern',0,128,0,0,0,99,99,0);
+INSERT INTO `mob_groups` VALUES (18,3128,67,'Phantom_Puk',0,128,0,650,0,75,75,0);
```
Comment (WinterSolstice8) [inline sql/mob_groups.sql:3274]: [17051655] = {['id']=17051655, ['name']="PhantomPuk", ['index']=7, ['level']=76},
    [17051656] = {['id']=17051656, ['name']="PhantomPuk", ['index']=8, ['level']=76},
    [17051657] = {['id']=17051657, ['name']="PhantomPuk", ['index']=9, ['level']=76},
    [17051658] = {['id']=17051658, ['name']="PhantomPuk", ['index']=10, ['level']=76},
    [17051659] = {['id']=17051659, ['name']="PhantomPuk", ['index']=11, ['level']=76},

Widescan seems to think these are all 76 (but I think that widescan is not perfectly accurate)

File: scripts/actions/mobskills/boreas_mantle.lua [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+            SpawnMob(cloneId):updateEnmity(hateTarget)
+            local cloneHP = math.floor(clone:getMaxHP() * phantomHPP) / 100
+            clone:setMod(xi.mod.DMG, 20000) -- Clones take triple damage
+            clone:setHP(cloneHP)
+        end
+    end
+
+    -- Handle Despawn of clones every 30 seconds
+    mob:timer(30000, function(mobArg)
+        for clone = mobArg:getID() + 1, mobArg:getID() + 4 do
```
Comment (Xaver-DaRed) [inline scripts/actions/mobskills/boreas_mantle.lua]: `for clone = phantomPuk + 1, phantomPuk + 4 do`

EDIT: Use the one bellow

Comment (WinterSolstice8) [reply scripts/actions/mobskills/boreas_mantle.lua]: That won't be in scope anymore

Comment (Xaver-DaRed) [reply scripts/actions/mobskills/boreas_mantle.lua]: I dunnow what you mean by scope, however, we dont need to fetch the original puk id several times.
Worst case scenario
```lua
    -- Handle Despawn of clones every 30 seconds
    mob:timer(30000, function(mobArg)
        local mobArgId = mobArg:getID()
        for clone = mobArgId + 1, mobArgId + 4 do
```

Comment (WinterSolstice8) [reply scripts/actions/mobskills/boreas_mantle.lua]: yeah that works

**Comment by CatsEyeXI** [comment]
> google drive link is for "call to arms"

fixed

File: sql/mob_groups.sql (hunk 1 above)
Comment (CatsEyeXI) [reply sql/mob_groups.sql:3274]: fixed

File: scripts/actions/mobskills/boreas_mantle.lua (hunk 2 above)
Comment (CatsEyeXI) [reply scripts/actions/mobskills/boreas_mantle.lua]: fixed

**Comment by CatsEyeXI** [comment]
image
is this on me?

**Comment by TeoTwawki** [comment]
> is this on me?

nope. seems to be fixed now. occasionally CI fails to get one of its dependencies.

**Review by Xaver-DaRed — APPROVED** [review]

**Comment by claywar** [comment]
This only covers area 1.  You're going to need areas 2 and 3 for this to work properly in all situations (mob data and armoury crates)

**Comment by CatsEyeXI** [comment]
> This only covers area 1.  You're going to need areas 2 and 3 for this to work properly in all situations (mob data and armoury crates)

There's a TODO to capture the positions of the objects in the other two battlefields. I don't have the resources to force myself into each one in retail. Won't the players just get spit out by if battlefield 1 is already occupied by someone? 

EDIT: I can make a really good approximation as to the spawn points in the other two battlefields if you're ok with that, I just don't know how to capture all three without using 3 characters. Thanks

**Comment by TeoTwawki** [comment]
> > This only covers area 1.  You're going to need areas 2 and 3 for this to work properly in all situations (mob data and armoury crates)
> 
> There's a TODO to capture the positions of the objects in the other two battlefields. I don't have the resources to force myself into each one in retail. Won't the players just get spit out by if battlefield 1 is already occupied by someone?
> 
> EDIT: I can make a really good approximation as to the spawn points in the other two battlefields if you're ok with that, I just don't know how to capture all three without using 3 characters. Thanks

if you can get 1 position from anything at the other arenas you can kind of cheat by offsetting your existing positions by that much. usually something like 600 yalm apart

**Pull request merged**